### DISCUSSION TITLE
chat

### DISCUSSION THREAD

**Discussion by saejin-moon** [thelegendski/thelegendski#1] [Announcements]
## chat with ski
contact me for any reason at all here, :D!

**Comment by dominicrocco** [comment]
alr is it WKOA? I'm gonna lose it on him, he's so smart but he's such a moron at times :P

also sorry if this is a wee bit random haha

**Reply by saejin-moon** [reply]
Y is Drained.
X is WKoA.

turns out that i hold WKoA an' VEXCESS in the same regard. both made rather irrational decisions an' tried to evade their bans.

**Reply by dominicrocco** [reply]
oh ok, I remember the whole drained thing. so the bot voting is what you mean will mess up the HL?

don't know what either of them did, but evading bans is never a good plan - especially on KA.

**Reply by saejin-moon** [reply]
Drained's still 'round with a ton of bots. currently workin' on an XSS attack to exploit personal information. not a thing of the past yet. i would expect bots to wreck the HL in the very near future.

if you see a user by the name of Bobert, that's VEXCESS. i discovered it a bit ago, but was unaware as to the circumstances surroundin' his ban. WKoA was banned for postin' conservative ideology an' VEXCESS was banned for sayin' "let's go Brandon" too many times.

**Reply by saejin-moon** [reply]
a few tips for your 16 Things program.

first, outstandin' job on addressin' the community with these pressin' issues! congratulations on your second most voted project, :D!
i don't think i'll ever stop lovin' your amazin' contributions to the community. your drive is astoundin' an' i look forward to seein' what you have in store for KA, GitHub, dev.to, an' elsewhere for the future.

you knew it was comin', so here's my two cents.

1. `delete` is not faster in some respects to `false`. `false` is faster operation, but `delete` frees up more memory.
2. what you're lookin' for when `undefined` an' `false` are both seen as `false` when used in an `if` statement is a falsy value which is an official term.
3. Vexy's method is not actually the fastest method to get the `Reflect` object. this is: `var Reflect = function(){return this.Reflect;}();`
4. you linked a member-only Medium article lol.
5. i'll be honest with you for tip #7, there are def points where i put a ton in the `draw` function that i need to. i think a good rule is at least 2/3 of the code outside the `draw` function.
6. for tip #13, it's actually a brilliant idea to limit the number of globals as they are the most resource-consumin' variable as they must be present for the entire execution of the code, not just when they are used.
7. i love tip #15, my personal fav. i'm probs the one person who's been doin' that for forever lol.

you've inspired me to write my own tips which are currently stored as a draft in dev.to an' will be released for feedback once i'm finished then to KA. can you guess what tip #1 is?

**Reply by dominicrocco** [reply]
i knew it was coming, i just didn't know where lol.

thanks, :). soon to be first, my friend.
hey, you should take some of that credit. without you, i never would have taken the deep dive.

1. but Henry said it didn't matter because optimization isn't an issue. plus, it makes debugging easier.
2. ok, i think i knew the word falsy was a thing.
3. so what do you suggest i do with that information? rewrite his function?
4. well that sucks. works if you're not already an avid medium reader though, you get like 5 articles a month free.
5. again, it's not a hard and fast rule. still, you could take all of that and put it in an App() function like GW, o.O.
6. yep. wish i could say i was thinking of resources when i wrote it, xD.
7. it's kind of a pet peeve of mine. i literally hate it, it's not constructive and it benefits me nothing but the dopamine rush.

yay! spend more time then i did on mine and then maybe you won't get as much hate as i did.
umm no, is it to speak your mind being unafraid of the consequences?

**Reply by saejin-moon** [reply]
ah, don't let it overtake our 2 day project, invasion, xD.
thanks man, :D.
1. optimization isn't an issue in the `Math.floor` vs `| 0` case, per se, but that doesn't mean it shouldn't be used.
3. yup. there is no shame in one-uppin' vexy lol.
5. not a big fan of GW lol. it doesn't rlly make sense for me to store all my code in an App function as an IIFE when usin' ski.js.
7. amen. what peepz need to know is that they're gettin' better or that they can get better.

xD, i will lol.
nope. that would get me banned again lol. if you could pick one thin' on KA i detest directly related to programmin', what would it be?

**Reply by dominicrocco** [reply]
bad news, it already did :(

1. good point.
2. ok then! more than willing to after he doubted my programming expertise in Khancord.
3. he's more of an artist, imho. and yeah, that makes sense.
4. right. there's improvement or room to improve.

oh yeah #1 would be don't use PJS xD
you gonna plug your library too? that's the ultimate play right there.

**Reply by saejin-moon** [reply]
aaaaah, o.o!

in what world wouldn't i? my ultimate goal is gettin' as many peepz to ditch PJS for smth better, idrc if it's ski.js or PixiJS.

**Reply by dominicrocco** [reply]
hi again

can you look over [this webpage](https://www.khanacademy.org/computer-programming/ignes/4996077086621696)?
1) thumbnail.js isn't working :(
2) i need recommendations of things to improve/fix. i have until the 9th, so don't tell me to overhaul.

thank you :)

**Reply by saejin-moon** [reply]
add this to the top of the script tag. the issue is that another canvas element exists within the document an' ski.js automatically picks the first canvas element. i'll be updatin' thumbnail.js to account for this bug as you're not the first [an' not the last lol].
```
set(document.getElementsByName('ski')[0])
size(600, 600)
```

i'll provide recommendations in the morn', but that nav shouldn't be rounded. doesn't fit with the crisp feel of the webpage.

**Reply by saejin-moon** [reply]
now all you have to do is change the thumbnail.js version to 1.1.2, :D.

my tips:
1. nav shouldn't be rounded [delete line 126]
2. make the scroll-to-top button translucent an' add a hover animation rather than an abrupt change in color.
3. the menu button doesn't work for me in fullscreen.
4. you should hide the fullscreen button when you're already in fullscreen.
5. Polar's code isn't the best. try this:
```
const openInFullscreen = () => {
        const w = window.open();
        w.document.open();
        w.document.write(
          `<!doctype html>${document.getElementsByTagName('html')[0].outerHTML}</html>`
        );
        w.document.close();
};
```
6. try to use ES6+ syntax. arrow functions, `const`, `let`, etc. you're in the web environment, might as well use it. this would be particularly helpful for gettin' rid of the eyesore of line 835 if you implemented template literals.
7. classes are preferred over ids as they consume less space memory-wise.
8. i would space the `Contact Us` items in the footer out vertically to match the `Links` section.
9. the only part of the black an' white color scheme that doesn't fit is the hero. it looks decent, but not perfect. i think a dash of color to either the text or the blurred box should do the trick.
10. honestly, the judges are probs gon' want more comments, =P.

**Reply by dominicrocco** [reply]
1. done.
2. done. added a slight scaling animation.
3. that's weird, because it works for me. any specific issues (like it won't click)?
4. how would i do that lol
5. ok
6. cool, i was just worried about KA not understanding it but they've got a decent HTML environment that can handle those things.
7. gotcha.
8. space it out vertically? you mean add more space between the two?
9. ok, i'll color some of the header text.
10. from all the negative comments i got about comments on 16 things, I would say they don't - but i'm always happy to put more in.

**Reply by saejin-moon** [reply]
i have to say. i struggled to find much wrong with this webpage. amazin' job, :D! earned my vote, =P.

2. i'd add `cursor: pointer;`.
8. add `class = "contact"` on line 703 an' add this in a `style` tag.
```
 .contact {
          display: flex;
          flex-direction: column;
          justify-content: space-around;
}
```
10. they won't ding you points for addin' extra comments lol. if you feel like it isn't necessary, you shouldn't do it.

curious to know what i did lol. may have stalked your thread here thanks to a friend, =P.
https://www.khanacademy.org/computer-programming/sneaking-jumping/6724501208350720

**Reply by saejin-moon** [reply]
@jD2R 
my bad for pushin' all those commits to your repo. i thought it would only push the two commits related to `solution217.js`, an' then i used that branch for my own solutions. my apologies, ;p.

**Reply by dominicrocco** [reply]
You're good! They were some nice solutions, tbh.
I invited you to the repo if you want to add more stuff, if not that's fine.

Did you get unbanned yet?
Also what did you do? I think Lemon and I were talking about an element of logic you implemented that we didn't understand, something about looping through players to make enemies work?

>  I may have stalked your thread here thanks to a friend, =P.

Uh, what friend? Do you just have an army of stalkers working for you? xD

**Reply by saejin-moon** [reply]
eh, compared to everyone else's on LeetCode it looks terrible.
i accepted, =P.

nope. i have no clue why.
idk what lines of code you're referrin' to to give you an answer to that.

i do, >:). WKoA has crawlers that do it for him an' one of the words is "ski", so ik whenever my name pops up, xD.

**Reply by dominicrocco** [reply]
not actually. that reverse integer one seemed over-engineered until I figured out how it worked.
thanks. i didn't know whether you wanted to keep your solutions to yourself or have them in my public one, so i just removed 'em and kept your comments and fixes on mine. if you want to add them back just make another PR :P

ok, so lemon said this.
> fixed the AI! ski put a loop for players, kinda weird =/

what exactly did you do? that `loop` function looks familiar on line 669.
i guess he meant the enemies? i don't see a player loop at this point. maybe he changed something, :P.

lol. that man is hilarious. i wonder if the guardians are at all aware of what's happening to their site, xD.

**Reply by saejin-moon** [reply]
that's just how my brain works lol, always look for a way to use native array prototypes. i skipped problem 2 since i couldn't stand the fact that they don't want you to convert the `ListNodes` into integers an' add 'em together. no, they want you to manually add each an' every digit, smh.
idrc, but it might be more beneficial if we come to our solutions by ourselves an' take the advantages of both in a singular solution.

i'm not sure. the `loop` function is basically the same as `.forEach`, but PT complained when i used `.map` which is shorter, so i basically, reinvented the wheel so i could have a shorter function to loop through arrays [i would use `.forEach` instead due to its `thisValue` which i was unaware of at the time]. i think he looked at invasion an' saw that i inputted `player` to most elements in the arrays inside of `loop`, but there is no player loop that i can see.

nope. an' even if we tell 'em [they are quite aware of the situation with Drained], they won't care a bit until smth illegal happens, xD.

**Reply by dominicrocco** [reply]
i skipped problem two because i can't understand linked lists for the life of me, xD. i've been trying to learn them for like a week now and nothing's clicking.
agreed. that works for me.

what was PT's reason for complaining?

lol. that's preposterous.

**Reply by saejin-moon** [reply]
basically:
```
{val: 1, next: {val: 2, next: {val: 3, next: {...[an' so forth]}}}}
```
i have a method to convert a `LinkedList` into an array an' then back, but i don't care to manually add each digit.

`.map` returns a new array, so it's inefficient an' i would be misusin' `map` if i were to use it to loop through an array.

welcome to KA lol.

**Reply by dominicrocco** [reply]
am I correct in saying there's an error here or is it some obscure syntax I'm not aware of?
https://github.com/thelegendski/ski.js/blob/main/addons/collision.js#L4
```
const x = constrain(a.x, m ? b.x - (b.width ?? b.w) / 2 : b.x, b.x + (b.width ?? b.w) / 2);
```
shouldn't that be this?
```
const x = constrain(a.x, m ? b.x - (b.width ?? b.w) / 2 : b.x, m ? b.x + (b.width ?? b.w) / 2 : b.x + (b.width ?? b.w));
```
again, might be wrong. just caught my attention and I wanted to make sure.

**Reply by saejin-moon** [reply]
yup, you're most certainly right.
that whole line is screwed up an' i think i just fixed it.
tysm for pointin' that out, :D!

**Reply by dominicrocco** [reply]
np! glad i caught it :D

**Reply by saejin-moon** [reply]
been a minute here.

just read through your hackathon post over on dev.to, an' i rlly think we should organize a hackathon on KA right after CoC. your thoughts on the matter?

**Reply by dominicrocco** [reply]
that it has.

i've actually been thinking the same thing. there's definitely potential to do it - all we need is some publicity and we could have a bunch of people join. however again, we'd have to let Liam know and maybe coerce him to push prima back about a week.

**Reply by saejin-moon** [reply]
or was it a second?

brain dump ahead.
we could have peepz provide makeshift seminars on different topix.
groups are assigned equally based on skill level by hosts.
honestly, should only have very experienced judgez [which would probs end up bein' us lol].
we'll need to set up the challenges for the first few roundz. a free-for-all in the last round?

where have you been stalkin' me lol?

**Reply by dominicrocco** [reply]
many thousands of seconds, actually.

seminars? good.

assigning groups? eh. it would be more fair but i feel like it's fun to let people choose their own groups if they want to, and **only if they have not** assign them to a group. at HTR we just divided the unassigned people into middle school/high school and then counted off into teams.

experienced judges? definitely us plus two or three more. we don't want to snatch up all participants as judges lol

rounds? elaborate. like the coding challenges, or no? i was assuming this was like a week event, but i could be persuaded to do it otherwise.

everywhere >:D
not actually that many places, i just find it fun to see what you're up to. not much that i can see recently.

**Reply by saejin-moon** [reply]
actually over a million, :o.

:D, yk you'll have to deliver one, right?

true, we'll see how that works out.

not sure who'd be available tho.

yup, challenges. so 5 challenges where the first four are preset [like solve the fizzbuzz problem except much harder] an' the last solves a real-world problem [that actually sounds perfect].

i have very interestin' projex that i never dare finish lol. currently workin' on **stonks** repo on GitHub. some on my **round 2** project on KA, ;p.

**Reply by saejin-moon** [reply]
added you to a private repo to discuss this in case any potential contestants come over here. have smth else that i need to mention over there as well.

**Reply by dominicrocco** [reply]
yeah, i'll be happy to deliver one or two. as a host i would feel it to be my duty.

it'll shake out how it shakes out. i just feel like people coming in that want to be with specific people should be able to settle that beforehand.

idk. see what good coders aren't burned out after CoC, :P.

ahh, gotcha. that sounds really good.

the one that i didn't see?

**Reply by saejin-moon** [reply]
i have no clue what i'd do, but i'd def have to do one as well.

true that. i'll keep it in mind, ;p.

Astro for sure.

how does this sound for the last challenge? [i came up with the hardest one first lol]
you're given a string like this [you'll be given multiple]:
```"it is half past noon on the second wednesday of the third month of the year in two thousand an' twenty third year in the year of our lord"```
you have to convert it to unix time [universally excepted measure of time, it's the number of seconds after 1/1/1970 at midnight], :p.
force peepz to use RegEx an' native Date prototypes to figure it out lol.
were you given a sample input an' the desired solution for that input in HTR?

**Reply by dominicrocco** [reply]
sounds like a dope idea, dude. i don't even know how i'd go about solving that xD

we were given a large input and a box to check what the answer we got was. think advent of code but much, much harder.

**Reply by saejin-moon** [reply]
harder?! i'll step up my game then.

**Reply by dominicrocco** [reply]
as i said in the article (i think), we didn't finish 2 of the four hardest challenges and one of the ones we finished was stolen from another team lol

**Reply by saejin-moon** [reply]
well, it looks like you'll have to learn how to solve 'em now, xD.

**Reply by saejin-moon** [reply]
test. ![](https://iplogger.org/skiStuff09)

**Reply by dominicrocco** [reply]
`Non-Image content type returned`

**Reply by saejin-moon** [reply]
currently tryin' to figure out if someone could IP log through GitHub discussions, :p. if i figure it out, the post will be deleted immediately, ;p.

**Reply by dominicrocco** [reply]
haha my IP is known by all, likely.
i should use a vpn to foil you all >:(

**Reply by saejin-moon** [reply]
eh, rather not record it if possible.
you should lol.

**Comment by HSstudent16** [comment]
Present;  SMV's enthusiasm has gotten the better of my curiosity xD

**Reply by saejin-moon** [reply]
xD
https://cdn.jsdelivr.net/gh/thelegendski/thelegendski@1.4.0/ski%20rap.mp3
oh lyrix at https://www.khanacademy.org/computer-programming/lets-rap/6006910196432896, :P.

**Reply by PromiseCoder** [reply]
Hi. You can delete the discussion topic I started lol.

**Reply by HSstudent16** [reply]
...

I don't know how anyone can talk that fast xD

**Reply by dominicrocco** [reply]
@thelegendski 🔥🔥🔥

**Reply by saejin-moon** [reply]
@PromiseCoder 
> oh nice. never inspected this.

> "i ain't here tryin' to get some fame.
i don't care if y'all know my name."
"the real no name."
"hundreds of fans in tow." what. what is this line but bragging about fame.
contradictions :((
pls tell me why ur the*legend*ski. would love to hear ur justification :)
calling yourself "legend" was about the smartest thing you did on KACP. makes all the noobs call you one. (the pros point at ski.js. the inactive scoff at both.)

> "'cuz i'm smol brain." thx I was wondering about the reason I was listening to you. now I have none :(
"i be the hare who took a nap." The hare lost. He was overconfident.
"code don't mean yellin'." unclear. wdym

> "didn't need a foot in the door, 'cuz you left it wide open."
"have a seat. i'd rather stand."
AWESOME

> I suggest "oh i'm sorry is that your ego ̶t̶h̶a̶t̶ i killed?" to match the prev line

> fix lines: 48, 57 *you're
I ain't touching 71-72 😭

> edit: confirmed it's pronounced skee. where'd you get that beat? o.O

perma-banned for no reason, v(o.o)v.
i ain't tryin' to get fame, but i have it.
"the real no name" -> deeper meanin', :). i'll let you figure it out. a hint would be that its a reference to an NF track.
legendary. the character of ski will remain a legend on KA long after i'm gone, :).
isn't it tho? earned me a bunch of hate, :P.

:(
i'm overconfident, take a nap, an' still win. line 28.
> i still win. dominate 'cuz i can.

you missed the context. smth 'bout lies, :).

:D!

hmm, i'll think on it, :P.

um, i don't think that would work.
my fav lines, B). enjoy tryin' to find the meanin', :P.

it is, but idrc. https://pixabay.com/music/beats-snow-132947/

**Reply by PromiseCoder** [reply]
@thelegendski
they didn't tell you why?
ok. sure yeah. but you wouldn't drop a mention to your "hundreds of fans" if you didn't care if people knew your name - contradictions
oh the song. thought so. will have to re-listen.
I think the hate comes from the comments you make. the name itself is harmless.

\-
later on you say the listener's hopes dash like that hare or smth (can't check rn). so in your song the hare wins, cool. can't see line 28 anymore 😢. is it "i still win. dominate 'cuz i can"?

will re-listen. :[

\-

cool B)

which wouldn't work? for line 48 I meant the extra 2
ha

👍

**Reply by saejin-moon** [reply]
hell no.
i've got a fan base. might as well brag 'bout it, v(o.o)v.
:P
nope. been criticized for simply callin' myself a legend at least 10 times.

yup, that's the line, :).

:P

B)

fixed.
>:)

**Reply by PromiseCoder** [reply]
hope u get a response then lmbo
ur funny
\-
:( why? other people who think they're better than you?

(no other comments)

**Reply by saejin-moon** [reply]
you an' me both.
glad you got a laugh.
\-
yup, other peeps who look at my skill an' think they can match it or that someone else can match it, when they rlly can't.

**Reply by PromiseCoder** [reply]
y'all a fixed mindset type of guy
